File: Logic & Design/Visual Studio Community Edition/Menu-Driven Programming/C#/Algorithm1/algorithm1.cs
/*
 * Programmer:          Corey Jenkins
 * Date:                February 11, 2023
 * Filename:            algorithm1.cs
 * Purpose:            A demonstration of an event-driven program using a if/else statement. 
*/

using System;

namespace Algorithm1
{
    class algorithm1
    {
        static void Main(string[] args)
        {
            menu();
            Console.ReadKey();
        }

        static void menu()
        {
            int menuSelection;
            Console.WriteLine("This is the main menu.");
            Console.WriteLine("1. Open the menu.");
            Console.WriteLine("2. Close the current document.");
            Console.WriteLine("3. Print the current document.");
            Console.WriteLine("4. Exit the program.");

            Console.WriteLine("Enter your selection:");
            menuSelection = Convert.ToInt32(Console.ReadLine());

           if(menuSelection == 1)
            {
                Console.WriteLine("Opening the document.");
            }
           else if(menuSelection == 2)
            {
                Console.WriteLine("Closing the document.");
            }
           else if(menuSelection == 3)
            {
                Console.WriteLine("Printing the document.");
            }
           else if(menuSelection == 4)
            {
                Console.WriteLine("Exiting the program.");
            }
           else
            {
                Console.WriteLine("An error has occurred.");
            }
        }
    }
}

File: Logic & Design/Visual Studio Community Edition/Menu-Driven Programming/C++/Algorithm1/Algorithm1.cpp
/*
* Programmer:		Corey Jenkins
* Date:				January 22, 2023
* Filename:			Algorithm1.cpp
* Purpose:			An algorithm involving input validation using if/else statements.
*/

#include <iostream>
#include <string>
using namespace std;

void Menu()
{
	int MenuSelection;
	cout << "This is the main menu." << endl;
	cout << "1. Open a menu." << endl;
	cout << "2. Close the current document." << endl;
	cout << "3. Print the current document. " << endl;
	cout << "4. Exiting the program." << endl;

	cout << "Enter your selection: " << endl;
	cin >> MenuSelection;

	if (MenuSelection == 1)
	{
		cout << "Opening the document." << endl;
	}
	else if (MenuSelection == 2)
	{
		cout << "Closing the document." << endl;
	}
	else if (MenuSelection == 3)
	{
		cout << "Printing the document." << endl;
	}
	else if (MenuSelection == 4)
	{
		cout << "Exiting the program." << endl;
	}
	else
	{
		cout << "That is an invalid selection." << endl;
	}
}

int main()
{
	Menu();
	system("pause");
	return 0;
}


